timer/src/main/java/com/toggl/timer/suggestions/domain/CalendarSuggestionProvider.kt

```diff
@@ -14,23 +13,26 @@ import kotlin.math.abs
 @Singleton
 class CalendarSuggestionProvider @Inject constructor(
     private val timeService: TimeService,
-    private val calendarService: CalendarService,
     private val dispatcherProvider: DispatcherProvider,
     @MaxNumberOfCalendarSuggestions private val maxNumberOfSuggestions: Int
 ) : SuggestionProvider {
     private val lookBackTimeSpan = Duration.ofHours(1)
     private val lookAheadTimeSpan = Duration.ofHours(1)
 
-    override suspend fun getSuggestions(suggestionsState: SuggestionsState): List<Suggestion> = withContext(dispatcherProvider.io) {
-        val now = timeService.now()
-        val startOfRange = now - lookBackTimeSpan
-        val endOfRange = now + lookAheadTimeSpan
+    override suspend fun getSuggestions(suggestionsState: SuggestionsState): List<Suggestion> =
+        withContext(dispatcherProvider.io) {
+            val now = timeService.now()
+            val startOfRange = now - lookBackTimeSpan
+            val endOfRange = now + lookAheadTimeSpan
+            val calendarEvents = suggestionsState.calendarEvents
 
-        calendarService
-            .getCalendarEvents(startOfRange, endOfRange, calendarService.getAvailableCalendars())
-            .filter { it.description.isNotBlank() }
-            .sortedBy { abs(now.minutesUntil(it.startTime)) }
-            .map { Suggestion.Calendar(it, suggestionsState.user.defaultWorkspaceId) }
-            .take(maxNumberOfSuggestions)
-    }
+            calendarEvents.values
+                .asSequence()
+                .filter { it.description.isNotBlank() }
+                .filter { it.startTime >= startOfRange && it.startTime <= endOfRange }
```

Comment:
Let's use a range here
```suggestion
                .filter { it.startTime in startOfRange..endOfRange }
```